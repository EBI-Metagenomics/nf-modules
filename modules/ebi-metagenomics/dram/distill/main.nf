process DRAM_DISTILL {
    tag "$meta.id"
    label 'process_high'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/dram:1.3.5--pyhdfd78af_0':
        'quay.io/biocontainers/dram:1.3.5--pyhdfd78af_0' }"

    containerOptions {
        if (workflow.containerEngine == 'singularity') {
            return "--bind ${moduleDir}/tests/fixtures/dram_distill_dbs/:/data/ --bind ${moduleDir}/tests/fixtures/dram_distill_dbs/CONFIG:/usr/local/lib/python3.10/site-packages/mag_annotator/CONFIG"
        } else {
            return "-v ${moduleDir}/tests/fixtures/dram_distill_dbs/:/data/ -v ${moduleDir}/tests/fixtures/dram_distill_dbs/CONFIG:/usr/local/lib/python3.10/site-packages/mag_annotator/CONFIG"
        }
    }

    input:
    tuple val(meta), path(tsv_input)

    output:
    tuple val(meta), path("*_dram.html"), emit: html
    tuple val(meta), path("*_dram.tsv") , emit: tsv
    path "versions.yml"                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '1.3.5' // WARN: dram has no option to print the tool version. This is the container version

    """
    DRAM.py \\
        distill \\
        -i ${dram_input}  \\
        -o dram_out \\
        ${args}
    mv dram_out/product.html ${prefix}_dram.html
    mv dram_out/product.tsv ${prefix}_dram.tsv
    
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        dram: $VERSION
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '1.3.5'
    """
    touch ${prefix}_dram.html
    touch ${prefix}_dram.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        dram: $VERSION
    END_VERSIONS
    """
}