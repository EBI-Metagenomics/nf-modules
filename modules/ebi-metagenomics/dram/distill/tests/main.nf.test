nextflow_process {

    name "Test Process DRAM_DISTILL"
    script "../main.nf"
    process "DRAM_DISTILL"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "dram_distill"

    test("test_dram_distill") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', prefix:"ERR4968564"], // meta map
                    file("${moduleDir}/tests/fixtures/full_input.tsv", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}