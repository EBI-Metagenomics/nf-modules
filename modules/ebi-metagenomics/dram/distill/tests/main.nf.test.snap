{
    "test_dram_distill": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "prefix": "ERR4968564"
                        },
                        "test_dram.html:md5,ca0e6e137649d70865168568ab92b10f"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "prefix": "ERR4968564"
                        },
                        "test_dram.tsv:md5,1b3861355147dc83ad6afd67d0258fff"
                    ]
                ],
                "2": [
                    "versions.yml:md5,474145e9ff22e5243cb59230b6f981ea"
                ],
                "distill_out_html": [
                    [
                        {
                            "id": "test",
                            "prefix": "ERR4968564"
                        },
                        "test_dram.html:md5,ca0e6e137649d70865168568ab92b10f"
                    ]
                ],
                "distill_out_tsv": [
                    [
                        {
                            "id": "test",
                            "prefix": "ERR4968564"
                        },
                        "test_dram.tsv:md5,1b3861355147dc83ad6afd67d0258fff"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,474145e9ff22e5243cb59230b6f981ea"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-06-25T17:42:39.304240761"
    }
}